itou/www/companies_views/views.py

```diff
@@ -68,28 +65,61 @@ def get_data_inclusion_services(code_insee):
             settings.API_DATA_INCLUSION_TOKEN,
         )
         try:
-            services = client.search_services(code_insee)
+            raw_services = client.search_services(code_insee)
         except DataInclusionApiException:
             # 15 minutes seems like a reasonable amount of time for DI to get back on track
             cache.set(cache_key, [], 60 * 15)
             return []
 
-        services = [s for s in services if s["modes_accueil"] == ["en-presentiel"]]
-        results = random.sample(services, min(len(services), 3))
-        results = [
-            r
-            | {
-                "thematiques_display": {displayable_thematique(t) for t in r["thematiques"]},
-                "dora_service_redirect_url": reverse(
-                    "companies_views:dora_service_redirect",
-                    kwargs={
-                        "source": r["source"],
-                        "service_id": r["id"],
-                    },
-                ),
-            }
-            for r in results
-        ]
+        services = []
+        for s in raw_services:
+            if "en-presentiel" not in s["modes_accueil"]:
+                continue
+            s["thematiques_display"] = {displayable_thematique(t) for t in s["thematiques"]}
+            s["dora_service_redirect_url"] = reverse(
+                "companies_views:dora_service_redirect",
+                kwargs={
+                    "source": s["source"],
+                    "service_id": s["id"],
+                },
+            )
+            services.append(s)
+
+        random.shuffle(services)
+        first_svc, second_svc, third_svc = None, None, None
+        department = code_insee[:2]
+        if department in ["59", "67"]:
+            for svc in services:
+                if svc["source"] == "soliguide":
+                    first_svc = svc
+                    services.remove(svc)
+                    break
+        if not first_svc and services:
+            first_svc = services.pop(0)
+
+        if services:
+            for svc in services:
+                if set(svc["thematiques_display"]) != set(first_svc["thematiques_display"]):
```

Comment:
Est-ce que tu ne veux pas un service qui a au moins un thématique différente ? 
Il faudrait des services à plusieurs thématiques dans `API_TEST_SERVICES`

Ça pourrait a priori se résoudre par une soustraction (est-ce qu'il y a au moins une nouvelle thématique par rapport à celles déjà présentes).

Reply:
En effet, je pense que je pourrais simplement faire ça, ça fera moins d'itérations.

Je vais également modifier API_TEST_SERVICES.